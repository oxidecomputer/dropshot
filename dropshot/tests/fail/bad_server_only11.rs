// Copyright 2024 Oxide Computer Company

#![allow(unused_imports)]

// Test for a trait with where clauses.

#[dropshot::server]
trait MyServer
where
    usize: std::fmt::Debug,
{
    type Context;
}

enum MyImpl {}

// This should not produce errors about the trait or the context type being
// missing.
impl MyServer for MyImpl {
    type Context = ();
}

fn main() {
    // These items will NOT be present because of the invalid trait, and will
    // cause errors to be generated.
    my_server::api_description::<MyImpl>();
    my_server::stub_api_description();
}
