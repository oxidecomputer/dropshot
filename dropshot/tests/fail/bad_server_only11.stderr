error: server `MyServer` must not have a where clause
  --> tests/fail/bad_server_only11.rs:9:1
   |
9  | / where
10 | |     usize: std::fmt::Debug,
   | |___________________________^

error[E0425]: cannot find function `api_description` in module `my_server`
  --> tests/fail/bad_server_only11.rs:26:16
   |
26 |     my_server::api_description::<MyImpl>();
   |                ^^^^^^^^^^^^^^^ not found in `my_server`

error[E0425]: cannot find function `stub_api_description` in module `my_server`
  --> tests/fail/bad_server_only11.rs:27:16
   |
27 |     my_server::stub_api_description();
   |                ^^^^^^^^^^^^^^^^^^^^ not found in `my_server`
